server/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java

```diff
@@ -267,7 +267,7 @@ protected void doExecute(Task task, Request request, ActionListener<Response> li
         private final Map<String, List<ShardRouting>> nodeIds;
         private final AtomicReferenceArray<Object> responses;
         private final AtomicInteger counter = new AtomicInteger();
-        private List<NoShardAvailableActionException> unavailableShardExceptions = new ArrayList<>();
+        private int unavailableShardCount = 0;
```

Comment:
@GrandFisher do you mind making `unavailableShardCount` final to be consistent with the other fields?
Unavailable shards could be counted with a local variable and then assigned to this field in the end of the constructor.

Reply:
@idegtiarenko  Thanks for your review, I'll fix this

Reply:
@idegtiarenko  I have fixed this and please review again. By the way, how can I improve the code to be able to pass the test, can you give me some advice ?

Reply:
May I rebase the master and git push force ?

Reply:
Please merge the latest master into your branch.
This should trigger another build

Reply:
@idegtiarenko  It seems pass all code checks, thanks for your advice.